libbeat/service/service.go

```diff
@@ -72,11 +74,49 @@ func BeforeRun() {
 	if *httpprof != "" {
 		go func() {
 			logp.Info("start pprof endpoint")
-			logp.Info("finished pprof endpoint: %v", http.ListenAndServe(*httpprof, nil))
+			mux := http.NewServeMux()
+
+			// register pprof
+			mux.HandleFunc("/debug/pprof/", pprofNet.Index)
```

Comment:
Do we need to take over all the handlers here? Is there a way to only replace the one we modify? The part I worry is if new handlers are added in golang, we have to keep track of it manually.